scipy/optimize/slsqp/slsqp_optmz.f

```diff
@@ -504,7 +505,7 @@ SUBROUTINE slsqpb (m, meq, la, n, x, xl, xu, f, c, g, a, acc,
 
   255 CONTINUE
       IF ((ABS(f-f0).LT.tol .OR. dnrm2_(n,s,1).LT.tol) .AND. h3.LT.tol
-     *     .AND. .NOT. badlin)
+     *     .AND. .NOT. badlin .AND. f .EQ. f)
```

Comment:
Not sure how fortran handles associativity.
is this line:

```
AND NOT (badlin AND f EQ f)
```
or
```
AND (NOT badlin) AND f EQ f
```
?

Reply:
The latter